src/main/java/org/vietabroader/controller/SheetFreshController.java

```diff
@@ -46,56 +48,68 @@ public SheetFreshController setColumnArray(String column, JTextField txt) {
+            if (currentState.getStatus() == GlobalState.Status.CONNECTED) {
 
-            String selected = (String) comSheets.getSelectedItem();
-            spreadsheet.setSheetName(selected);
-            logger.info("Sheet " + selected + " is selected");
+                VASpreadsheet spreadsheet = currentState.getSpreadsheet();
 
-            int from = Integer.parseInt(txtRowFrom.getText());
-            int to = Integer.parseInt(txtRowTo.getText());
-            spreadsheet.setRow(from, to);
-            logger.info("Row " + from + " " + to);
+                spreadsheet.removeAllColumns();
 
-            for(String key: columnArray.keySet()) {
-                spreadsheet.setColumnChar(key, columnArray.get(key).getText());
-                logger.info(key + " " + columnArray.get(key).getText());
-            }
+                String selected = (String) comSheets.getSelectedItem();
+                spreadsheet.setSheetName(selected);
+                logger.info("Sheet " + selected + " is selected");
+
+                int from = Integer.parseInt(txtRowFrom.getText());
+                int to = Integer.parseInt(txtRowTo.getText());
+                spreadsheet.setRow(from, to);
+                logger.info("Row " + from + " " + to);
 
-            (new RefreshWorker(lblSheetMessage)).execute();
+                for(String key: columnArray.keySet()) {
+                    spreadsheet.setColumnChar(key, columnArray.get(key).getText());
+                    logger.info(key + " " + columnArray.get(key).getText());
+                }
+
+                (new RefreshWorker()).execute();
+                prgIndicator.setIndeterminate(true);
+            }
         });
     }
 
-    private static class RefreshWorker extends SwingWorker<String, Object> {
-        private JLabel lbl;
+    private class RefreshWorker extends SwingWorker<VASpreadsheet, Object> {
 
-        RefreshWorker(JLabel lbl) {
-            this.lbl = lbl;
+        @Override
+        protected VASpreadsheet doInBackground() throws Exception {
+            VASpreadsheet spreadsheet = GlobalState.getInstance().getSpreadsheet();
+            spreadsheet.refreshAllColumns();
+            return spreadsheet;
         }
 
         @Override
-        protected String doInBackground() throws Exception {
-            GlobalState currentState = GlobalState.getInstance();
-            VASpreadsheet spreadsheet = currentState.getSpreadsheet();
-            logger.info("Start to refresh the sheet" + spreadsheet.getSpreadsheetTitle());
+        protected void done() {
             try {
+                VASpreadsheet spreadsheet = GlobalState.getInstance().getSpreadsheet();
                 spreadsheet.refreshAllColumns();
+
+                long now = Calendar.getInstance().getTimeInMillis();
+                lblSheetMessage.setText("The sheet has been refreshed at " + new Timestamp(now));
+                logger.info("The sheet has been refreshed");
```

Comment:
After fixing the bug with `label`, try temporarily adding this code after this line (remember to delete this afterward):
```
  for (int i = 0; i < 10; i++) {
            logger.info("Key, " + i + ": " + spreadsheet.readValue("Key", i).toString());
   }
```
Then try connecting to this sheet: 
`https://docs.google.com/spreadsheets/d/1BxiMVs0XRA5nFMdKvBdBZjgmUUqptlbs74OgvE2upms/edit#gid=0`

After refreshing (row should go from 1 to 10 or more), you should see some content in the sheet appearing in the log.